File: java/Decorator_Pattern/decorator/Whip.java
package java.Decorator_Pattern.decorator;

import beverage.Beverage;

public class Whip extends CondimentDecorator {
    public Whip(Beverage beverage){
        this.beverage = beverage;
    }

    public String getDescription(){
        return beverage.getDescription() + ", Whip";
    }

    public double cost(){
        return beverage.cost() + .20;
    }

}

File: problems/coffee-machine/whip.py
from condiment_decorator import CondimentDecorator

class Whip(CondimentDecorator):
    def get_description(self) -> str:
        return self.beverage.get_description() + ", Whip"

    def get_cost(self) -> float:
        return self.beverage.get_cost() + .30